src/authentication/json-api-authorization-spec.ts

```diff
@@ -462,5 +536,51 @@ describe('authorization for api routes', () => {
         expect(api.getActivityHandler).to.not.have.been.called;
       });
     });
+    describe('getPreviewHandler', () => {
+      it('should allow fetching the preview for a deployment in an authorized project', async () => {
+        // Arrange
+        const { server, authentication, api } = await arrange(
+          'userHasAccessToDeployment',
+          true,
+          'getPreviewHandler',
+        );
+        // Act
+        await makeRequest(server, '/preview/1-1?sha=12345678', 'GET');
+
+        // Assert
+        expect(authentication.userHasAccessToDeployment).to.have.been.calledOnce;
+        expect(api.getPreviewHandler).to.have.been.calledOnce;
+      });
+      it('should not allow fetching the preview for a deployment in an unauthorized project', async () => {
+        // Arrange
+        const { server, authentication, api } = await arrange(
+          'userHasAccessToDeployment',
+          false,
+          'getPreviewHandler',
+        );
+        // Act
+        const response = await makeRequest(server, '/preview/1-1?sha=12345678', 'GET');
+
+        expect(response.statusCode).to.eq(401);
+        expect(authentication.userHasAccessToDeployment).to.have.been.calledOnce;
+        expect(api.getPreviewHandler).to.not.have.been.called;
+      });
+      it('should allow fetching the preview for an open deployment without authentication', async () => {
+        // Arrange
+        const { server, authentication, api } = await arrange(
+          'isOpenDeployment',
+          true,
+          'getPreviewHandler',
+        );        // Act
+        await server.inject({
+          method: 'GET',
+          url: 'http://foo.com/preview/1-1?sha=12345678',
+        });
+        // Assert
+        expect(authentication.isOpenDeployment).to.have.been.calledOnce;
+        expect(api.getPreviewHandler).to.have.been.calledOnce;
+      });
+    });
+
```

Comment:
Should tests also be added for cases where there's an open deployment and the user has access credentials?